server/src/services/mod.rs

```diff
@@ -6,3 +6,7 @@ pub mod get_updates;
 pub mod move_file;
 pub mod new_account;
 pub mod rename_file;
+
+pub fn username_is_valid(username: String) -> bool {
```

Comment:
You should borrow here, then callers don't need to `clone`. If you really need to mutate internally you can `clone` yourself. But instead you could probably write this in a way that doesn't require you to create a new string.